ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -450,6 +459,97 @@ func checkResponse(path string, code int, body []byte) error {
 	return fmt.Errorf("IPFS-post '%s' unsuccessful: %d: %s", path, code, body)
 }
 
+// postCtxStreaming makes a POST request against
+// the ipfs daemon, reads the body of the response line by line
+// at evey millisecond. It cancels the context if timeout duration
+// has passed since last new data on the buffer.
+func (ipfs *Connector) postCtxStreaming(ctx context.Context, path string, timeout time.Duration) error {
```

Comment:
this needs refactoring. a `doPostCtx` can happen in `Pin` and then pass the response reader to either a `handleRefsProgress` or `handlePinProgress` (or better name, but two separate functions).

Reply:
This would create quite some duplicate code, but I guess we are trading that with readability, will do.